ios/FluentUI/Badge Field/BadgeView.swift

```diff
@@ -351,9 +382,20 @@ open class BadgeView: UIView {
 
     open override func sizeThatFits(_ size: CGSize) -> CGSize {
         let labelSize = label.sizeThatFits(CGSize(width: CGFloat.infinity, height: CGFloat.infinity))
-        let width = UIScreen.main.roundToDevicePixels(labelSize.width) + self.size.horizontalPadding * 2
+        let labelWidth = UIScreen.main.roundToDevicePixels(labelSize.width)
+        let labelHeight = UIScreen.main.roundToDevicePixels(labelSize.height)
+
+        let width: CGFloat
+        let height: CGFloat
+        if let avatarSize = dataSource?.avatar?.state.size.size {
+            width = labelWidth + avatarSize + self.size.horizontalPadding * 3
+            height = max(avatarSize, labelHeight) + self.size.avatarVerticalPadding * 2
```

Comment:
Validation suggestion:
The avatar grows in height and width if it has its ring enabled. Have you experimented changing the state object to have the ring updated and see if it will still be within the acceptable margins and bounds of the badge?

Reply:
No need for custom view now

Reply:
This is not related to any custom view change. This is about the Avatar updating its frame when state properties are changed.

Reply:
As I tried, frame won't update when `avatar.state.isRingVisible = true` called.
But frame will be update when call
```
avatar.state.isRingVisible = true
badge?.invalidateIntrinsicContentSize()
```